mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxPathDAOImpl.java

```diff
@@ -202,10 +201,10 @@ private CassandraIdAndPath fromRowToCassandraIdAndPath(Row row) {
             .setString(MAILBOX_NAME, mailboxPath.getName()));
     }
 
-    public CompletableFuture<Stream<CassandraIdAndPath>> readAll() {
-        return cassandraAsyncExecutor.execute(selectAll.bind())
-            .thenApply(cassandraUtils::convertToStream)
-            .thenApply(stream -> stream.map(this::fromRowToCassandraIdAndPath));
+    public Flux<CassandraIdAndPath> readAll() {
+        return cassandraAsyncExecutor.executeReactor(selectAll.bind())
+            .flatMapMany(Flux::fromIterable)
```

Comment:
didn't you have a concern about using Flux::fromIterable (about thread safety I guess)?

Reply:
I see it many times in this PR

Reply:
No, `fromIterable` is ok, the problem we have is when we use `toIterable`